### PR TITLE
10.0 pos ms issues 3

### PR DESCRIPTION
**Pull request by KolushovAlexandr**

### CODE REVIEW

File: pos_multi_session_restaurant/static/src/js/pos_multi_session_restaurant.js [hunk 1]
```diff
@@ -101,6 +101,11 @@ odoo.define('pos_multi_session_restaurant', function(require){
             var floor_table = false;
             var order = false;
 
+            // Multi session without floors
+            if (!this.config.ms_floor_ids.length) {
+                return PosModelSuper.prototype.ms_create_order.call(this, options);
+            }
```
Comment (yelizariev) [inline pos_multi_session_restaurant/static/src/js/pos_multi_session_restaurant.js:107]: seems ok

File: pos_multi_session_sync/models/pos_multi_session_sync_models.py [hunk 2]
```diff
@@ -60,7 +60,8 @@ def prepare_new_session(self, message):
         self.write({
             'run_ID': run_ID,
         })
-        old_orders = self.env['pos_multi_session_sync.order'].search([('state', '=', 'draft'),
+        old_orders = self.env['pos_multi_session_sync.order'].search([('multi_session_ID', '=', self.multi_session_ID),
+                                                                      ('state', '=', 'draft'),
```
Comment (yelizariev) [inline pos_multi_session_sync/models/pos_multi_session_sync_models.py:64]: :+1:

File: pos_multi_session_sync/models/pos_multi_session_sync_models.py [hunk 3]
```diff
@@ -165,7 +166,7 @@ def set_order(self, message):
             order = order.create({
                 'order': json.dumps(message),
                 'order_uid': order_uid,
-                'multi_session_ID': self.id,
+                'multi_session_ID': self.multi_session_ID,
```
Comment (yelizariev) [inline pos_multi_session_sync/models/pos_multi_session_sync_models.py:169]: :+1:

File: pos_multi_session_sync/models/pos_multi_session_sync_models.py [hunk 4]
```diff
@@ -180,12 +181,19 @@ def get_sync_all(self, message):
...
-                .create({'multi_session_ID': self.multi_session_ID, 'pos_ID': pos_ID, 'user_ID': user_ID})
+            if same_id_poses and len(same_id_poses) == 1:
+                # pos.config was assigned to a different multi session
+                same_id_poses.multi_session_ID = self.multi_session_ID
+                pos = same_id_poses
+            else:
+                # in case there were already created plenty same_id_poses (before the fix)
+                same_id_poses.unlink()
+                pos = self.env['pos_multi_session_sync.pos'] \
+                    .create({'multi_session_ID': self.multi_session_ID, 'pos_ID': pos_ID, 'user_ID': user_ID})
```
Comment (yelizariev) [inline pos_multi_session_sync/models/pos_multi_session_sync_models.py]: Looks complicated. Can we simplify it? May be
```

pos = self.env['pos_multi_session_sync.pos'].search([("pos_ID", "=", pos_ID)])

if len(pos) > 1:
    # workaround for deployment with a bug
   # TODO: remove this in odoo 13 
    pos.unlink()
    pos = None

if not pos:
     pos = ...create(...)

```

File: pos_multi_session_sync/models/pos_multi_session_sync_models.py [hunk 5]
```diff
@@ -180,12 +181,19 @@ def get_sync_all(self, message):
...
-                  .search([('multi_session_ID', '=', self.multi_session_ID), ("pos_ID", "=", pos_ID)])
+        same_id_poses = self.env['pos_multi_session_sync.pos'].search([("pos_ID", "=", pos_ID)])
+        pos = same_id_poses.filtered(lambda p: p.multi_session_ID == self.multi_session_ID)
         run_ID = message['data']['run_ID']
         if not pos:
-            pos = self.env['pos_multi_session_sync.pos'] \
-                .create({'multi_session_ID': self.multi_session_ID, 'pos_ID': pos_ID, 'user_ID': user_ID})
+            if same_id_poses and len(same_id_poses) == 1:
+                # pos.config was assigned to a different multi session
+                same_id_poses.multi_session_ID = self.multi_session_ID
```
Comment (yelizariev) [inline pos_multi_session_sync/models/pos_multi_session_sync_models.py]: Is it possible, that ``same_id_poses.multi_session_ID != self.multi_session_ID``?

File: pos_multi_session_sync/models/pos_multi_session_sync_models.py [hunk 6]
```diff
@@ -205,7 +213,7 @@ def get_sync_all(self, message):
             orders.append(msg)
         else:
             for order in self.env['pos_multi_session_sync.order'] \
-                             .search([('multi_session_ID', '=', self.id), ('state', '=', 'draft'),
+                             .search([('multi_session_ID', '=', self.multi_session_ID), ('state', '=', 'draft'),
```
Comment (yelizariev) [inline pos_multi_session_sync/models/pos_multi_session_sync_models.py:215]: :+1:

**Review by yelizariev — CHANGES_REQUESTED** [review]

File: pos_multi_session_sync/models/pos_multi_session_sync_models.py (hunk 5 above)
Comment (KolushovAlexandr) [reply pos_multi_session_sync/models/pos_multi_session_sync_models.py]: yes if `pos.config was assigned to a different multi session`

Comment (KolushovAlexandr) [reply pos_multi_session_sync/models/pos_multi_session_sync_models.py]: before those changes it created a new record with the same `pos_ID` and different `multi_session_ID`

Comment (yelizariev) [reply pos_multi_session_sync/models/pos_multi_session_sync_models.py]: I mean that same_id_poses is the same record as self.

Comment (yelizariev) [reply pos_multi_session_sync/models/pos_multi_session_sync_models.py]: ok, I got it. self is not a record of ``pos_multi_session_sync.pos``

**Review by yelizariev — APPROVED** [review]

**Review by ilmir-k — APPROVED** [review]

**Pull request merged**